File: content/post/2021-11-03-sublime-text-for-data-science/test-build.jl
using DataFrames, GLM
using Random, Distributions
using Plots

# simulated data
n = 100;
x = range(0, 10, length=n);
y = (x .- 4).^2 .+ 2 + rand(Normal(0, 1), n);

# data frame
X = DataFrame(x=x, y=y);

# linear regression model
mod = lm(@formula(y ~ x + x^2), X);

# predictions
yhat = predict(mod);

# plot measurements and predictions
scatter(x, y, axis="x values", yaxis="y values",
        color="blue", label="Measurement")
plot!(x, yhat, color="red", label="Prediction")
plot!(legend=:topleft)
display(plot!())

File: content/post/2021-11-03-sublime-text-for-data-science/test-build.R
library("tidyverse")

# simulated data
n <- 100
x <- seq(0, 10, length.out = n)
y <- (x - 4)^2 + 2 + rnorm(n)

# data frame
X <- data.frame(y = y, x = x)

# linear regression model
mod <- lm(y ~ x + I(x^2), data = X)

# predictions
yhat <- predict(mod)

# plot measurements and predictions
colors = c('Measurement' = 'blue',
           'Prediction' = 'red')
p <- ggplot() +
    geom_point(data = X, aes(x = x, y = y,
                             color = 'Measurement')) +
    geom_line(aes(x = x, y = yhat,
                  color = 'Prediction')) +
    xlab('x values') +
    ylab('y values') +
    scale_color_manual(name = '', values = colors) +
    theme(legend.position = 'top')
print(p)
